repository_service_tuf/cli/admin/ceremony.py

```diff
@@ -31,28 +31,45 @@
     _online_settings_prompt,
     _print_root,
     _root_threshold_prompt,
-    _warn_no_save,
 )
+from repository_service_tuf.helpers.api_client import (
+    URL,
+    bootstrap_status,
+    send_payload,
+    task_status,
+)
+from repository_service_tuf.helpers.tuf import save_payload
+
+DEFAULT_PATH = "ceremony-payload.json"
 
 
 @admin.command()  # type: ignore
 @click.option(
-    "--save",
-    "-s",
+    "--out",
     is_flag=False,
-    flag_value="ceremony-payload.json",
-    help="Write json result to FILENAME (default: 'ceremony-payload.json')",
+    flag_value=DEFAULT_PATH,
+    help=f"Write output json result to FILENAME (default: '{DEFAULT_PATH}')",
     type=click.File("w"),
 )
-def ceremony(save) -> None:
+@click.pass_context
+def ceremony(context: Any, out: Optional[click.File]) -> None:
     """Bootstrap Ceremony to create initial root metadata and RSTUF config."""
     console.print("\n", Markdown("# Metadata Bootstrap Tool"))
+    settings = context.obj["settings"]
+    # Running online ceremony requires connection to the server and
+    # confirmation that the server is indeed ready for bootstap.
+    if not settings.get("SERVER") and not out:
+        raise click.ClickException(
+            "Either '--api-sever'/'SERVER' in RSTUF config or '--out' needed"
+        )
 
-    if not save:  # pragma: no cover
-        _warn_no_save()
+    if settings.get("SERVER"):
+        bs_status = bootstrap_status(settings)
+        if bs_status.get("data", {}).get("bootstrap") is True:
+            raise click.ClickException(f"{bs_status.get('message')}")
```

Comment:
Do we need to check the bootstrap status here? Wouldn't it be enough, if the server was ready for bootstrap at the time of sending the bootstrap message below? We need to handle the error case, where the server wasn't ready, anyway, because things can change between here and line 124.

IMO adding this LBYL check here is premature ux optimisation and I suggest to remove it.

Reply:
I think it's better with the check that without it.
It's better to receive a message that bootstrap has been done already:
<img width="954" alt="image" src="https://github.com/repository-service-tuf/repository-service-tuf-cli/assets/16246778/fa85e832-4010-4868-928a-4c1d5ffcef97">

as compared to doing all of the work for the ceremony, adding at least two keys, and signing with one of them just to get:
<img width="738" alt="image" src="https://github.com/repository-service-tuf/repository-service-tuf-cli/assets/16246778/ff98c1ae-2ec5-4335-9e55-466edd9d80b0">

Reply:
As I said, we should handle the case (or rather the many possible cases), where the server fails, anyway. This includes giving proper user feedback and likely dumping the result to a local file so that the work is not lost.

Adding this preliminary check but not handling errors is the wrong order of priorities.

Reply:
We already handle other server problems in `bootstrap_status`:
<img width="593" alt="image" src="https://github.com/repository-service-tuf/repository-service-tuf-cli/assets/16246778/ea2a4cdf-234e-4319-8c23-063c37368466">

Reply:
I am talking about `send_payload`. It could fail for all sorts of reasons.  If that happens, the user's work will be lost, unless you  handle it.

Reply:
We handle errors in `send_payload` as well:
![image](https://github.com/repository-service-tuf/repository-service-tuf-cli/assets/16246778/6bd9ed86-661f-4afa-ada4-534f52924d47)

Reply:
Yes, you re-raise them and work will be lost.

Reply:
Ahaa, I see now what you mean.
I agree saving the payload in the default path if not saved before sending to the payload sounds smart.
Will add this here as it's a small addition.

Reply:
Not sure, if we're really on the same page. I was trying to say that the preliminary check (`bootstrap_status`) only gives you a limited certainty that posting the data (`send_payload`) will succeed. So I was actually suggesting to reduce functionality and not to add more. :)

Reply:
I don't agree that it's an unnecessary check.
Calling `bootstrap_status` early can help the user when:
1. the given `settings.SERVER` is not reachable
2. RSTUF deployment has been bootstrapped already

For `1` he will be able to resolve his `--api-url`/`settings.SERVER` before thinking about the ceremony as a whole.
For `2` if we don't check that in the beginning and he does a full ceremony even if we save payload locally it's likely that the ceremony was a waste of time. As probably it doesn't make sense to restart a bootstrap RSTUF deployment.
Also, if `2` happens at the beginning of the ceremony then the user will be able to assess whether restarting RSTUF is required or not.

If you want to discuss it further maybe it's good to do it on a daily when it's three of us.

Reply:
Some notes from a conversation with @lukpueh and @kairoaraujo:
- if we want to have a pre-check before running the ceremony we need to take into account other failures besides those checked now. For example, the server could become unreachable when we finish the ceremony
- one option to consider is to start a server session at the beginning of the ceremony which will be used in the end when sending the payload.

We have decided we would remove this check for now and an issue about this will be created.
Here is the issue: https://github.com/repository-service-tuf/repository-service-tuf-cli/issues/618